src/cdk-experimental/combobox/combobox.ts

```diff
@@ -143,6 +143,13 @@ export class CdkCombobox<T = unknown> implements OnDestroy, AfterContentInit {
 
     if (valueChanged) {
       this.panelValueChanged.emit(value);
+      this._setTextContent(value);
+    }
+  }
+
+  private _setTextContent(content: T) {
+    if (typeof content === 'string') {
```

Comment:
I would probably not check if it's exactly a string and just do ``textContent = `${content}`;``. We could then later add support for custom toString function.